File: Stack/stack.cpp
#include <vector>

using namespace std;

template <class T>
class stack {
private:
	vector<T> items;
	int size;

public:
	void push(T);
	T pop();
	int getSize();
	T top();
};

template <class T>
void stack<T>::push(T item) {
	items.push_back(item);
	size++;
}

template <class T>
T stack<T>::pop() {
	if (size == 0) {
		throw "No items to pop";
	}
	else {
		T temp = items[size - 1];
		items.pop_back();
		size--;
		return temp;
	}
}

template <class T>
int stack<T>::getSize() {
	return size;
}

template <class T>
T stack<T>::top() {
	return items[size - 1];
}

File: Stack/Stack.kt
class Stack<T:Comparable<T>>(list:MutableList<T>) {

    var items: MutableList<T> = list


    fun isEmpty():Boolean = this.items.isEmpty()

    fun count():Int = this.items.count()

    fun push(element:T) {
        val position = this.count()
        this.items.add(position, element)
    }

    override  fun toString() = this.items.toString()

    fun pop():T? {
        if (this.isEmpty()) {
            return null
        } else {
            val item =  this.items.count() - 1
            return this.items.removeAt(item)
        }
    }

    fun peek():T? {
        if (isEmpty()) {
            return null
        } else {
            return this.items[this.items.count() - 1]
        }
    }

}